src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -216,7 +216,7 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                     case ConversionKind.Identity:
                     case ConversionKind.ImplicitReference:
                     case ConversionKind.Unboxing:
-                    case ConversionKind.NullLiteral:
+                    case ConversionKind.DefaultOrNullLiteral:
```

Comment:
This method called only from `BindDeclarationPattern` (and more specifically when there is a problem with the type in the declaration pattern), so I don't think this line is needed for either `null` or `default`.